lib/rexml/parsers/baseparser.rb

```diff
@@ -635,6 +645,9 @@ def process_instruction(start_position)
           @source.position = start_position
           raise REXML::ParseException.new(message, @source)
         end
+        if @tags.size.zero? && !@root_tag.nil? && match_data[1] == "xml"
+          raise ParseException.new("Malformed XML: Extra XML declaration at the end of the document (got '<?#{match_data[1]}')", @source)
+        end
```

Comment:
XML declaration must be the first item. So we can use different condition for the XML declaration position check.
For example:

```ruby
if match_data[1] == "xml"
  raise ... if @document_status
  # ...
end
```

How about working on this in a separated PR?

Reply:
OK.
I see.
I will make another PR.
Thanks!

Reply:
I created https://github.com/ruby/rexml/pull/162